[v2.13.0] Add Configure GitHub App page (#2081)
rancher/rancher-docs: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-github-app.md

:::

Before configuring the provider in Rancher, you must first create a GitHub App for your organization, generate a client secret for GitHub App and generate a private key for your GitHub App. Refer to [Registering a GitHub App](https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app) for details.

### Create GitHub App

1. Open your [GitHub organization settings](https://github.com/settings/organizations).
1. To the right of the organization, select **Settings**.
1. In the left sidebar, click **Developer settings** > **GitHub Apps**.
1. Click **New Github App**.
1. Fill in the GitHub App configuration form with these values:

    - **GitHub App name**: Anything you like, e.g. My Rancher
    - **Application description**: Optional, can be left blank
    - **Homepage URL**: https://localhost:8443

>     - **Callback URL**: https://localhost:8443/verify-auth

Reviewer: ```suggestion
    - **GitHub App name**: Anything you like, e.g. `My Rancher`.
    - **Application description**: Optional, can be left blank.
    - **Homepage URL**: `https://localhost:8443`.
    - **Callback URL**: `https://localhost:8443/verify-auth`.
```

1. Select **Create Github App**.

### Generate a Client Secret

Generate a [client secret](https://docs.github.com/en/rest/authentication/authenticating-to-the-rest-api?apiVersion=2022-11-28#using-basic-authentication) on the settings page for your app.

1. Go to your GitHub App.
1. Next to **Client Secrets**, select **Generate a new client secret**.

### Generate a Private Key

> Generate a [private key](https://docs.github.com/en/enterprise-server@3.14/apps/creating-github-apps/authenticating-with-a-github-app/managing-private-keys-for-github-apps#generating-private-keys) on the settings page for your app.

Reviewer: Do we need to link to version 3.14 specifically?

Author: @btat thanks for catching this. I'll add https://docs.github.com/en/enterprise-server/apps/creating-github-apps/authenticating-with-a-github-app/managing-private-keys-for-github-apps#generating-private-keys instead.

1. Go to your GitHub App.
1. Next to **Private Keys**, click **Generate a private key**.

## GitHub App Auth Provider Configuration

To set up the GitHub App Auth Provider in Rancher, follow these steps:

1. Navigate to the **Users & Authentication** section in the Rancher UI.
1. Select **Auth Providers**.
1. Select the **GitHub App** tile.
1. Gather and enter the details of your GitHub App into the configuration form fields.

  | Field Name | Description |
  | ---------- | ----------- |
  | **Client ID** (Required) | The client ID of your GitHub App. |
  | **Client Secret** (Required) | The client secret of your GitHub App. |
  | **GitHub App ID** (Required) | The numeric ID associated with your GitHub App. |
  | **Installation ID** (Optional) | If you want to restrict authentication to a single installation of the App, provide its specific numeric Installation ID. |
  | **Private Key** (Required) | The contents of the Private Key file (in PEM format) generated by GitHub for your App. |

1. Select **Enable**. Rancher attempts to validate the credentials and, upon success, will activate the GitHub App provider.

> Once enabled, users logging in via the GitHub App provider will be automatically identified and you can leverage your GitHub Organization's teams and users to configure Role-Based Access Control (RBAC) and to assign permissions to projects and clusters.

Reviewer: ```suggestion
After its enabled, users logging in via the GitHub App provider are automatically identified and you can leverage your GitHub Organization's teams and users to configure Role-Based Access Control (RBAC) and to assign permissions to projects and clusters.
```
